### ISSUE TITLE
Constant disconnects

### ISSUE THREAD

**Issue by jaspervdm** [mimblewimble/grin#480]
I am running an up to date node on testnet1, and I have noticed that it keeps disconnecting from other nodes many many times. It reopens the connection but after some time (<1min usually) the connection drops again. It also keeps asking the nodes for blocks but never receives them. Multiple people in Gitter were seeing the disconnects. Here is a part of the log (I removed many of the "Requesting block" lines). In this part you can see for example it loses connection to 109.74.202.16:13414 multiple times.

Any look into this would be appreciated! I would be happy to provide more information if necessary. Thanks.

```
Dec 14 00:00:41.373 INFO Connected to peer PeerInfo { capabilities: FULL_HIST | UTXO_HIST | PEER_LIST | FULL_NODE, user_agent: "MW/Grin 0.1.20171212", version: 1, addr: V4(109.74.202.16:13414), total_difficulty: Difficulty { num: 138463500 } }
Dec 14 00:00:41.373 DEBG Saving newly connected peer 109.74.202.16:13414.
Dec 14 00:00:41.373 DEBG save_peer: V4(109.74.202.16:13414) marked Healthy
Dec 14 00:00:41.374 DEBG connect_and_req: ok. attempting send_peer_request
Dec 14 00:00:41.374 DEBG Asking 109.74.202.16:13414 for more peers.
Dec 14 00:00:41.375 DEBG Received 16 peer addrs, saving.
Dec 14 00:00:41.375 DEBG Received 14 peer addrs, saving.
Dec 14 00:00:41.384 DEBG Received 16 peer addrs, saving.
Dec 14 00:00:41.386 DEBG Received 14 peer addrs, saving.
Dec 14 00:00:41.386 DEBG handshake connect with nonce - 10620421931885922814, sender - V4(0.0.0.0:13414), receiver - V4(192.241.168.77:13414)
Dec 14 00:00:41.399 DEBG Received 10 peer addrs, saving.
Dec 14 00:00:41.416 DEBG Received 14 peer addrs, saving.
Dec 14 00:00:41.452 DEBG handshake connect with nonce - 12753090366647888368, sender - V4(0.0.0.0:13414), receiver - V4(24.7.36.144:13414)
Dec 14 00:00:41.468 DEBG Received 13 peer addrs, saving.
Dec 14 00:00:41.482 DEBG Received 13 peer addrs, saving.
Dec 14 00:00:41.671 INFO Connected to peer PeerInfo { capabilities: FULL_HIST | UTXO_HIST | PEER_LIST | FULL_NODE, user_agent: "MW/Grin 0.1.20171212", version: 1, addr: V4(192.241.168.77:13414), total_difficulty: Difficulty { num: 138463500 } }
Dec 14 00:00:41.671 DEBG Saving newly connected peer 192.241.168.77:13414.
Dec 14 00:00:41.671 DEBG save_peer: V4(192.241.168.77:13414) marked Healthy
Dec 14 00:00:41.672 DEBG connect_and_req: ok. attempting send_peer_request
Dec 14 00:00:41.672 DEBG Asking 192.241.168.77:13414 for more peers.
Dec 14 00:00:41.685 DEBG peer total_diff @ height (ping/pong): 104.236.150.29:13414: 135645922 @ 15032 vs us: 135645922 @ 15032
Dec 14 00:00:41.685 DEBG Received 13 peer addrs, saving.
Dec 14 00:00:41.720 DEBG peer total_diff @ height (ping/pong): 139.59.105.169:13414: 137567572 @ 15595 vs us: 135645922 @ 15032
Dec 14 00:00:41.721 DEBG Received 13 peer addrs, saving.
Dec 14 00:00:41.771 DEBG Received 15 peer addrs, saving.
Dec 14 00:00:41.907 DEBG peer total_diff @ height (ping/pong): 192.241.168.77:13414: 138463500 @ 16094 vs us: 135645922 @ 15032
Dec 14 00:00:41.956 INFO Connected to peer PeerInfo { capabilities: FULL_HIST | UTXO_HIST | PEER_LIST | FULL_NODE, user_agent: "MW/Grin 0.1.20171204", version: 1, addr: V4(24.7.36.144:13414), total_difficulty: Difficulty { num: 138463500 } }
Dec 14 00:00:41.956 DEBG Saving newly connected peer 24.7.36.144:13414.
Dec 14 00:00:41.956 DEBG save_peer: V4(24.7.36.144:13414) marked Healthy
Dec 14 00:00:41.956 DEBG connect_and_req: ok. attempting send_peer_request
Dec 14 00:00:41.957 DEBG Asking 24.7.36.144:13414 for more peers.
Dec 14 00:00:42.138 DEBG Received 18 peer addrs, saving.
Dec 14 00:00:42.362 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12, c0d20054, 4247981c, 3a834d2c, 013c5a4b, d1f3d56a, fd88ca63, dc940d7e, 27b87a10, 8d16606d, 654061d9, cd228f82, 8f1127c3, 5408a39f, eee135bd, 5476ee98, 5f3759b4, 1842f6e8, bac6a5f2, d89315b8, c1c423eb, b5e96d59, 502c6770, 659563a4, f52e4f56, 243a708c, a501b295, c2afcf3a, 08b3cdb5, 04ebb787, ec231dc7, 6f609506, e2aa052e, fadc0b15, 59110e7f, f47b1b45, ad29f74a, 3d304427, fb6bc108, 603fe7f4, 1d892b63, 060b6a69, 9959d8c3, 5edfe1b0, 76452ac7, 28cbdeaa, 20bdeb5e, adc3d4a7, 81c11151, be8adeb5, c059fad2] from 6 peers
Dec 14 00:00:42.413 DEBG Requesting block a50a64c6 from peer 37.59.48.28:13414 [[50 more]]
Dec 14 00:00:42.955 DEBG peer total_diff @ height (ping/pong): 83.162.208.57:13414: 138463500 @ 16094 vs us: 135645922 @ 15032
Dec 14 00:00:43.309 DEBG peer total_diff @ height (ping/pong): 51.15.219.12:13414: 32421152 @ 9861 vs us: 135645922 @ 15032
Dec 14 00:00:44.157 DEBG peer total_diff @ height (ping/pong): 24.7.36.144:13414: 138463500 @ 16094 vs us: 135645922 @ 15032
Dec 14 00:00:46.366 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:00:47.363 DEBG peer total_diff @ height (ping/pong): 164.132.230.217:13414: 35491469 @ 0 vs us: 135645922 @ 15032
Dec 14 00:00:47.429 TRCE Too long: Block Some(a50a64c6)
Dec 14 00:00:47.430 INFO Client 37.59.48.28:13414 connection lost: Timeout
Dec 14 00:00:47.430 TRCE Too long: Block Some(cada5ca9)
Dec 14 00:00:47.430 INFO Client 109.74.202.16:13414 connection lost: Timeout
Dec 14 00:00:47.483 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12, c0d20054, 4247981c, 3a834d2c, 013c5a4b, d1f3d56a, fd88ca63, dc940d7e, 27b87a10, 8d16606d, 654061d9, cd228f82, 8f1127c3, 5408a39f, eee135bd, 5476ee98, 5f3759b4, 1842f6e8, bac6a5f2, d89315b8, c1c423eb, b5e96d59, 502c6770, 659563a4, f52e4f56, 243a708c, a501b295, c2afcf3a, 08b3cdb5, 04ebb787, ec231dc7, 6f609506, e2aa052e, fadc0b15, 59110e7f, f47b1b45, ad29f74a, 3d304427, fb6bc108, 603fe7f4, 1d892b63, 060b6a69, 9959d8c3, 5edfe1b0, 76452ac7, 28cbdeaa, 20bdeb5e, adc3d4a7, 81c11151, be8adeb5, c059fad2] from 6 peers
Dec 14 00:00:47.483 DEBG Requesting block a50a64c6 from peer 37.59.48.28:13414. [[50 more]]
Dec 14 00:00:47.672 TRCE Too long: Block Some(538264d9)
Dec 14 00:00:47.672 INFO Client 192.241.168.77:13414 connection lost: Timeout
Dec 14 00:00:47.957 TRCE Too long: Block Some(f461135c)
Dec 14 00:00:47.957 INFO Client 24.7.36.144:13414 connection lost: Timeout
Dec 14 00:00:48.017 DEBG peer total_diff @ height (ping/pong): 188.166.171.24:13414: 138463500 @ 16094 vs us: 135645922 @ 15032
Dec 14 00:00:49.286 DEBG peer total_diff @ height (ping/pong): 104.236.150.29:13414: 135645922 @ 15032 vs us: 135645922 @ 15032
Dec 14 00:00:49.355 TRCE Too long: Block Some(3a78b375)
Dec 14 00:00:49.355 INFO Client 188.166.171.24:13414 connection lost: Timeout
Dec 14 00:00:49.356 TRCE Too long: Block Some(a695b21e)
Dec 14 00:00:49.356 INFO Client 83.162.208.57:13414 connection lost: Timeout
Dec 14 00:00:51.488 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:00:52.197 DEBG peer total_diff @ height (ping/pong): 46.101.149.191:13414: 135647553 @ 15033 vs us: 135645922 @ 15032
Dec 14 00:00:52.602 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12, c0d20054, 4247981c, 3a834d2c, 013c5a4b, d1f3d56a, fd88ca63, dc940d7e, 27b87a10, 8d16606d, 654061d9, cd228f82, 8f1127c3, 5408a39f, eee135bd, 5476ee98, 5f3759b4, 1842f6e8, bac6a5f2, d89315b8, c1c423eb, b5e96d59, 502c6770, 659563a4, f52e4f56, 243a708c, a501b295, c2afcf3a, 08b3cdb5, 04ebb787, ec231dc7, 6f609506, e2aa052e, fadc0b15, 59110e7f, f47b1b45, ad29f74a, 3d304427, fb6bc108, 603fe7f4, 1d892b63, 060b6a69, 9959d8c3, 5edfe1b0, 76452ac7, 28cbdeaa, 20bdeb5e, adc3d4a7, 81c11151, be8adeb5, c059fad2] from 6 peers
Dec 14 00:00:52.629 DEBG Requesting block a50a64c6 from peer 37.59.48.28:13414. [[50 more]]
Dec 14 00:00:52.975 DEBG extract_ip - 0.0.0.0, Ok(V4(83.162.208.57:40154))
Dec 14 00:00:52.975 DEBG Success handshake with 83.162.208.57:13414.
Dec 14 00:00:52.975 DEBG Saving newly connected peer 83.162.208.57:13414.
Dec 14 00:00:52.975 DEBG save_peer: V4(83.162.208.57:13414) marked Healthy
Dec 14 00:00:53.029 DEBG Got 13 peer addrs to send.
Dec 14 00:00:56.605 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:00:57.720 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12, c0d20054, 4247981c, 3a834d2c, 013c5a4b, d1f3d56a, fd88ca63, dc940d7e, 27b87a10, 8d16606d, 654061d9, cd228f82, 8f1127c3, 5408a39f, eee135bd, 5476ee98, 5f3759b4, 1842f6e8, bac6a5f2, d89315b8, c1c423eb, b5e96d59, 502c6770, 659563a4, f52e4f56, 243a708c, a501b295, c2afcf3a, 08b3cdb5, 04ebb787, ec231dc7, 6f609506, e2aa052e, fadc0b15, 59110e7f, f47b1b45, ad29f74a, 3d304427, fb6bc108, 603fe7f4, 1d892b63, 060b6a69, 9959d8c3, 5edfe1b0, 76452ac7, 28cbdeaa, 20bdeb5e, adc3d4a7, 81c11151, be8adeb5, c059fad2] from 6 peers
Dec 14 00:00:57.720 DEBG Requesting block a50a64c6 from peer 37.59.48.28:13414. [[50 more]]
Dec 14 00:00:59.209 DEBG peer total_diff @ height (ping/pong): 139.59.105.169:13414: 137567572 @ 15595 vs us: 135645922 @ 15032
Dec 14 00:01:01.269 DEBG peer total_diff @ height (ping/pong): 88.99.251.87:13414: 0 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:01.270 DEBG peer total_diff @ height (ping/pong): 94.130.66.48:13414: 0 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:01.277 DEBG peer total_diff @ height (ping/pong): 46.101.149.191:13414: 135647553 @ 15033 vs us: 135645922 @ 15032
Dec 14 00:01:01.279 DEBG peer total_diff @ height (ping/pong): 164.132.230.217:13414: 35491469 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:01.285 DEBG peer total_diff @ height (ping/pong): 83.162.208.57:13414: 138463500 @ 16094 vs us: 135645922 @ 15032
Dec 14 00:01:01.298 DEBG peer total_diff @ height (ping/pong): 51.15.219.12:13414: 32444953 @ 9881 vs us: 135645922 @ 15032
Dec 14 00:01:01.447 DEBG peer total_diff @ height (ping/pong): 104.236.150.29:13414: 135645922 @ 15032 vs us: 135645922 @ 15032
Dec 14 00:01:01.465 DEBG peer total_diff @ height (ping/pong): 139.59.105.169:13414: 137567572 @ 15595 vs us: 135645922 @ 15032
Dec 14 00:01:01.726 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:02.827 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12, c0d20054, 4247981c, 3a834d2c, 013c5a4b, d1f3d56a, fd88ca63, dc940d7e, 27b87a10, 8d16606d, 654061d9, cd228f82, 8f1127c3, 5408a39f, eee135bd, 5476ee98, 5f3759b4, 1842f6e8, bac6a5f2, d89315b8, c1c423eb, b5e96d59, 502c6770, 659563a4, f52e4f56, 243a708c, a501b295, c2afcf3a, 08b3cdb5, 04ebb787, ec231dc7, 6f609506, e2aa052e, fadc0b15, 59110e7f, f47b1b45, ad29f74a, 3d304427, fb6bc108, 603fe7f4, 1d892b63, 060b6a69, 9959d8c3, 5edfe1b0, 76452ac7, 28cbdeaa, 20bdeb5e, adc3d4a7, 81c11151, be8adeb5, c059fad2] from 6 peers
Dec 14 00:01:02.827 DEBG Requesting block a50a64c6 from peer 37.59.48.28:13414. [[50 more]]
Dec 14 00:01:02.852 DEBG Received block 4250d366 at 16095 from network, going to process.
Dec 14 00:01:02.852 DEBG pipe: process_block 4250d366 at 16095 with 0 inputs and 1 outputs.
Dec 14 00:01:02.852 DEBG pipe: validate_header cuckoo size 16
Dec 14 00:01:02.853 DEBG Block 4250d366 refused by chain: Orphan
Dec 14 00:01:02.956 DEBG peer total_diff @ height (ping/pong): 83.162.208.57:13414: 138464305 @ 16095 vs us: 135645922 @ 15032
Dec 14 00:01:02.956 TRCE Too long: Block Some(3a78b375)
Dec 14 00:01:02.956 INFO Client 83.162.208.57:13414 connection lost: Timeout
Dec 14 00:01:03.295 DEBG peer total_diff @ height (ping/pong): 51.15.219.12:13414: 32444953 @ 9881 vs us: 135645922 @ 15032
Dec 14 00:01:06.832 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:07.363 DEBG peer total_diff @ height (ping/pong): 164.132.230.217:13414: 35491469 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:07.955 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:08.003 DEBG Requesting block a50a64c6 from peer 83.162.208.57:13414. [[10 more]]
Dec 14 00:01:09.285 DEBG peer total_diff @ height (ping/pong): 104.236.150.29:13414: 135645922 @ 15032 vs us: 135645922 @ 15032
Dec 14 00:01:09.685 DEBG extract_ip - 0.0.0.0, Ok(V4(109.74.202.16:60710))
Dec 14 00:01:09.686 DEBG Success handshake with 109.74.202.16:13414.
Dec 14 00:01:09.686 DEBG Saving newly connected peer 109.74.202.16:13414.
Dec 14 00:01:09.686 DEBG save_peer: V4(109.74.202.16:13414) marked Healthy
Dec 14 00:01:09.751 DEBG Got 13 peer addrs to send.
Dec 14 00:01:11.272 DEBG monitor_peers: 2 most_work_peers, 13 connected, 319 total known
Dec 14 00:01:11.276 DEBG monitor_peers: all 319 = 13 healthy + 1 banned + 305 defunct
Dec 14 00:01:11.277 DEBG cleaning V4(188.166.171.24:13414), not connected
Dec 14 00:01:11.277 DEBG cleaning V4(83.162.208.57:13414), not connected
Dec 14 00:01:11.277 DEBG cleaning V4(24.7.36.144:13414), not connected
Dec 14 00:01:11.277 DEBG cleaning V4(37.59.48.28:13414), not connected
Dec 14 00:01:11.278 DEBG cleaning V4(192.241.168.77:13414), not connected
Dec 14 00:01:11.835 DEBG Got 13 peer addrs to send.
Dec 14 00:01:11.898 DEBG Got 13 peer addrs to send.
Dec 14 00:01:11.956 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:12.195 DEBG peer total_diff @ height (ping/pong): 46.101.149.191:13414: 135647553 @ 15033 vs us: 135645922 @ 15032
Dec 14 00:01:13.075 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:13.075 DEBG Requesting block a50a64c6 from peer 109.74.202.16:13414. [[10 more]]
Dec 14 00:01:17.076 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:18.012 DEBG Got 13 peer addrs to send.
Dec 14 00:01:18.166 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:18.166 DEBG Requesting block a50a64c6 from peer 109.74.202.16:13414. [[10 more]]
Dec 14 00:01:18.191 DEBG Got 13 peer addrs to send.
Dec 14 00:01:19.199 DEBG peer total_diff @ height (ping/pong): 139.59.105.169:13414: 137567572 @ 15595 vs us: 135645922 @ 15032
Dec 14 00:01:19.605 DEBG peer total_diff @ height (ping/pong): 109.74.202.16:13414: 138464305 @ 16095 vs us: 135645922 @ 15032
Dec 14 00:01:19.605 TRCE Too long: Block Some(a50a64c6)
Dec 14 00:01:19.606 INFO Client 109.74.202.16:13414 connection lost: Timeout
Dec 14 00:01:21.269 DEBG peer total_diff @ height (ping/pong): 88.99.251.87:13414: 0 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:21.269 DEBG peer total_diff @ height (ping/pong): 94.130.66.48:13414: 0 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:21.275 DEBG peer total_diff @ height (ping/pong): 46.101.149.191:13414: 135647553 @ 15033 vs us: 135645922 @ 15032
Dec 14 00:01:21.277 DEBG peer total_diff @ height (ping/pong): 164.132.230.217:13414: 35491469 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:21.446 DEBG peer total_diff @ height (ping/pong): 104.236.150.29:13414: 135645922 @ 15032 vs us: 135645922 @ 15032
Dec 14 00:01:21.465 DEBG peer total_diff @ height (ping/pong): 139.59.105.169:13414: 137567572 @ 15595 vs us: 135645922 @ 15032
Dec 14 00:01:22.168 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:23.264 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:23.283 DEBG Requesting block a50a64c6 from peer 109.74.202.16:13414. [[10 more]]
Dec 14 00:01:24.010 DEBG peer total_diff @ height (ping/pong): 51.15.219.12:13414: 32509889 @ 9911 vs us: 135645922 @ 15032
Dec 14 00:01:24.068 DEBG peer total_diff @ height (ping/pong): 51.15.219.12:13414: 32509889 @ 9911 vs us: 135645922 @ 15032
Dec 14 00:01:27.265 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:27.363 DEBG peer total_diff @ height (ping/pong): 164.132.230.217:13414: 35491469 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:28.387 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:28.387 DEBG Requesting block a50a64c6 from peer 109.74.202.16:13414. [[10 more]]
Dec 14 00:01:29.287 DEBG peer total_diff @ height (ping/pong): 104.236.150.29:13414: 135645922 @ 15032 vs us: 135645922 @ 15032
Dec 14 00:01:32.198 DEBG peer total_diff @ height (ping/pong): 46.101.149.191:13414: 135647553 @ 15033 vs us: 135645922 @ 15032
Dec 14 00:01:32.388 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:33.498 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:33.498 DEBG Requesting block a50a64c6 from peer 109.74.202.16:13414. [[10 more]]
Dec 14 00:01:37.500 DEBG body_sync: body_head - 563439a7, 15032, header_head - b30263f8, 24557, sync_head - b30263f8, 24557
Dec 14 00:01:38.604 DEBG block_sync: 15032/24557 requesting blocks [a50a64c6, a695b21e, cada5ca9, 68ffb996, 7e7e9bd7, 538264d9, 23e355b2, 3a78b375, f461135c, ff739e12] from 1 peers
Dec 14 00:01:38.605 DEBG Requesting block a50a64c6 from peer 109.74.202.16:13414. [[10 more]]
Dec 14 00:01:39.199 DEBG peer total_diff @ height (ping/pong): 139.59.105.169:13414: 137567572 @ 15595 vs us: 135645922 @ 15032
Dec 14 00:01:39.661 DEBG extract_ip - 0.0.0.0, Ok(V4(109.74.202.16:60716))
Dec 14 00:01:39.661 DEBG Success handshake with 109.74.202.16:13414.
Dec 14 00:01:39.662 DEBG Saving newly connected peer 109.74.202.16:13414.
Dec 14 00:01:39.662 DEBG save_peer: V4(109.74.202.16:13414) marked Healthy
Dec 14 00:01:39.708 DEBG Got 13 peer addrs to send.
Dec 14 00:01:41.272 DEBG monitor_peers: 1 most_work_peers, 8 connected, 319 total known
Dec 14 00:01:41.276 DEBG monitor_peers: all 319 = 13 healthy + 1 banned + 305 defunct
Dec 14 00:01:41.277 DEBG peer total_diff @ height (ping/pong): 88.99.251.87:13414: 0 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:41.277 DEBG peer total_diff @ height (ping/pong): 94.130.66.48:13414: 0 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:41.285 DEBG peer total_diff @ height (ping/pong): 46.101.149.191:13414: 135647553 @ 15033 vs us: 135645922 @ 15032
Dec 14 00:01:41.285 DEBG peer total_diff @ height (ping/pong): 164.132.230.217:13414: 35491469 @ 0 vs us: 135645922 @ 15032
Dec 14 00:01:41.294 DEBG peer total_diff @ height (ping/pong): 51.15.219.12:13414: 32525626 @ 9921 vs us: 135645922 @ 15032
Dec 14 00:01:41.298 DEBG peer total_diff @ height (ping/pong): 109.74.202.16:13414: 138464305 @ 16095 vs us: 135645922 @ 15032
```

**Comment by ignopeverell** [comment]
@antiochp has a fix that should help a lot with that in #477

**Comment by ignopeverell** [comment]
Hopefully a lot better now, feel free to reopen with more info if not.
